src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -197,11 +209,15 @@ def convert_pytorch_state_dict_to_flax(pt_state_dict, flax_model):
                 continue
 
             # also add unexpected weight so that warning is thrown
-            flax_state_dict[("params",) + flax_key] = jnp.asarray(flax_tensor)
+            flax_state_dict[("params",) + flax_key] = (
+                jnp.asarray(flax_tensor) if not is_bfloat_16 else jnp.asarray(flax_tensor, dtype=jnp.bfloat16)
+            )
```

Comment:
And then for here you can reuse the `weight_dtypes` dict to know if you need to convert back to bfloat16.